src/meltano/core/project_settings_service.py

```diff
@@ -61,6 +65,23 @@ def __init__(self, *args, config_service: ConfigService = None, **kwargs):
             **self.config_override,
         }
 
+        if self.get("project_id") is None:
```

Comment:
There is still some weirdness if a user manually adds the `project_id` key at the environment-level. These changes will result in it being added to `meltano.yml` at the top level (but not removed from the environment config it originated from). 

Does it make sense to have per-environment project IDs? If so we can clean up this issue by checking `self.project.active_environment is not None` here before editing the config on-disk. The consequence however would be that we don't restore the project ID from `analytics.json` if there is no active environment. I'm not sure if that can ever occur.

Reply:
Actually, instead of erroring, we should just rewrite the config file to move `project_id` up. An error will be raised if there are multiple that conflict, but if there's only one or they all match there's no need for an error.

Reply:
In the end I decided against erroring at all. As you say it doesn't make sense, so silently getting rid of the env-level project IDs seems fine if necessary.

There's currently a bug that I'm not sure is worth fixing where the project ID that gets brought up from the env-level to the top-level will be turned into a UUID via hashing if it isn't one already. This is because `ProjectSettingsService` is initialized multiple times, and on the first time it will not have an active environment. As a result the project ID is taken from `analytics.json` instead of the env-level config. Later when `ProjectSettingsService` is initialized with an active environment, the UUID from `analytics.json` will be preferred over the one at the env-level since it's now the top-level `project_id`.